### ISSUE TITLE
Windows Narrator cannot show correct Grid focused rectangle if the display scale is not 100%

### ISSUE THREAD

**Issue by petercoin** [microsoft/microsoft-ui-xaml#8720]
### Describe the bug

The focused rectangle area doesn't match the window size if a Grid container is used as root container. This issue also can be found by using Accessibility Insights for Windows by checking the size of the pane.

### Steps to reproduce the bug

1. Create a new Win UI 3 C++ packaged project.
2. Change the default container from StackPanel to Grid.
3. Run the application.
4. Click on a blank space of the application.
5. Press Ctrl+Win+Enter to execute Narrator.
6. You will see the rectangle area doesn't match the Window size if the display scale is not 100%.

### Expected behavior

The focused rectangle should be the same size as the window size.

### Screenshots

100%
![winui3_grid_narrator_scale_100](https://github.com/microsoft/microsoft-ui-xaml/assets/12906762/f0cc766d-a263-4171-bc07-76e743058020)

125%
![winui3_grid_narrator_scale_125](https://github.com/microsoft/microsoft-ui-xaml/assets/12906762/3f37b97e-85c2-4d0f-a56d-da90ab866d21)

### NuGet package version

WinUI 3 - Windows App SDK 1.3.3: 1.3.230724000

### Windows version

Windows 11 (22H2): Build 22621
